pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -78,11 +80,33 @@ type GarbageCollector struct {
 	absentOwnerCache *ReferenceCache
 
 	workerLock sync.RWMutex
+
+	// kcp
+	clusterName logicalcluster.Name
 }
 
 var _ controller.Interface = (*GarbageCollector)(nil)
 var _ controller.Debuggable = (*GarbageCollector)(nil)
 
+// NewClusterAwareGarbageCollector creates a new GarbageCollector for the provided logical cluster.
```

Comment:
This could go in a separate _patch.go file to minimize rebase conflicts - wdyt?

Reply:
Absolutely good idea! Re-pushed with 52fa5f286ecb1c1f273ef4ecd06043d5531d4d02 that move the factory method into the patch file.